File: cpp/scope-of-variable.cpp
#include <iostream>
using namespace std;

void f1(int num1);
void f2();
void f3();
void f4();

void pointer(int* num) {
    int* newPointer = new int(6);
    num = newPointer;
}

void f1(int num1) {
    num1++;
}

int main() {

    int* p = new int(12);
    pointer(p);
    cout << *p << endl;  // output 12

    int num1 = 4;
    cout << num1 << endl; // 4
    f1(num1);
    cout << num1 << endl; // 4

    f2();
    f3();

    f4();  // output 2
    f4();  // output 3

    return 0;
}

int y;

void f2() {
    y++;
}

void f3() {
    cout << y << endl;
}

void f4() {
    static int x = 1;
    x++;
    cout << x << endl;
}

File: python-from-scratch/scope-of-variable.py
def function1(num: int):
    num += 1

num1 = 4
print(num1)
function1(num1)
print(num1)


website = "JournalDev.com"

def print_website():
    global website
    print(f'My favorite website is {website}')

    website = 'Wikipedia.com'
    print(f'My favorite website is {website}')

print_website()
print(f'My favorite website is {website}')


import static_variable_pk as st

def static_test():
    st.static_var += 4
    print(st.static_var)

static_test()  # output 5
static_test()  # output 9
